src/Raven.Studio/typescript/models/database/index/indexDefinition.ts

```diff
@@ -122,20 +121,39 @@ class indexDefinition {
 
         this.reduce.extend({
             required: {
-                onlyIf: () => this.hasReduce()
-            }
+                onlyIf: () => this.hasReduce() && !this.reduce()
+            },
+            validation: [
+                {
+                    validator: (reduceContent: string) => {
+                        if (reduceContent && reduceContent.trim())
+                            return true; 
```

Comment:
validator: (reduceContent: string) => reduceContent && reduceContent.trim()

so we always return boolean value.

Reply:
done